tests/integration/piv_worker_test.py

```
@@ -0,0 +1,228 @@
+            "navitia_token": "dengdengdengdeng",
+            "connector_type": ConnectorType.piv.value,
+            "broker_url": self.broker_url,
+            "exchange_name": exchange_name,
+            "queue_name": queue_name,
+        }
+        response = self.test_client.post("/contributors", json=new_contrib)
+        assert response.status_code == 201
+
+        # Launch a PivWorker
+        self.thread = threading.Thread(target=launch_piv_worker, args=(self.pg_docker_fixture, contributor_id))
+        self.thread.start()
+        wait_until.call(lambda: fassert(self.thread.is_alive()))
+
+    def __exit__(self, type, value, traceback):
+        # Remove the contributor
+        self.test_client.delete("/contributors/{}".format(contributor_id))
+        # PivWorker should die eventually when no PIV contributors is available
+        wait_until.call(lambda: fassert(not self.thread.is_alive()))
+
+
+def test_mq_message_received_and_stored(
+    test_client, pg_docker_fixture, rabbitmq_docker_fixture, broker_connection, mq_handler
+):
+    with PivWorkerTest(test_client, rabbitmq_docker_fixture.url, pg_docker_fixture):
+        # Check that PivWorker is creating the queue
+        wait_until.call(lambda: fassert(is_queue_exists(broker_connection, queue_name)))
+
+        # Check that MQ message is received and stored in DB
+        mq_handler.publish(str('{"key": "Some valid JSON"}'), contributor_id)
+        wait_until.call(lambda: fassert(RealTimeUpdate.query.count() == 1))
+
+
+def test_exchange_changed(test_client, pg_docker_fixture, rabbitmq_docker_fixture, broker_connection):
+    with PivWorkerTest(test_client, rabbitmq_docker_fixture.url, pg_docker_fixture):
+        # Modify the 'exchange_name' in the contributor
+        new_exchange_name = "pivert"
+        modified_contrib = {
+            "id": contributor_id,
+            "exchange_name": new_exchange_name,
+        }
+        response = test_client.put("/contributors", json=modified_contrib)
```

Minor: to me the "happy path" is creating exchange outside first, then branching PivWorker.

Otherwise, I expect the worker to check is exchange exist and retry until it does.
So it's OK to test like that, but it tests a bit more than happy-path.

TODO (probably): separate discussion to see if it should be checked, tested and all.

---

Might be done in another PR. I believe the `PivWorker` is not resilient (and most likely fails silently) if the `Exchange` doesn't exist before-hand.

---

I did change the test though. See 648cd24adfbf45b0e701326013f32cd89dfe2c7a.